File: lib/model/content/content.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'content.freezed.dart';
part 'content.g.dart';

@freezed
abstract class Content with _$Content {
  const factory Content({
    required dynamic createdAt,
    required dynamic updatedAt,
    required String uid,
    required String contentId,
    required String storageIndexId,
    required List<String> refferedDocumentIds,
    required List<String> refferedDocumentFileNames,
    required String title,
    required String status,
    @Default(false) bool isPublic,
    @Default(false) bool isDeleted,
  }) = _Content;
  factory Content.fromJson(Map<String, dynamic> json) =>
      _$ContentFromJson(json);
}

File: functions/src/model/content.py
from pydantic import BaseModel
from datetime import datetime

class Content(BaseModel):
    createdAt: datetime
    updatedAt: datetime
    uid: str
    contentId: str
    storageIndexId: str
    refferedDocumentIds: list[str]
    refferedDocumentFileNames: list[str]
    title: str
    status: str
    isPublic: bool
    isDeleted: bool